sormas-app/app/src/main/java/de/symeda/sormas/app/pathogentest/edit/PathogenTestNewActivity.java

```diff
@@ -91,8 +103,16 @@ protected PathogenTest queryRootEntity(String recordUuid) {
 	@Override
 	protected PathogenTest buildRootEntity() {
 		// basic instead of reference, because we want to have at least the related person
-		Sample associatedSample = DatabaseHelper.getSampleDao().queryUuid(sampleUuid);
-		return DatabaseHelper.getSampleTestDao().build(associatedSample);
+
+		if (sampleUuid != null) {
+			Sample associatedSample = DatabaseHelper.getSampleDao().queryUuid(sampleUuid);
+			return DatabaseHelper.getSampleTestDao().build(associatedSample);
+		}
+		if (environmentSampleUuid != null) {
+			EnvironmentSample associatedEnvironmentSample = DatabaseHelper.getEnvironmentSampleDao().queryUuid(environmentSampleUuid);
+			return DatabaseHelper.getSampleTestDao().build(associatedEnvironmentSample);
+		}
+		throw new RuntimeException("Not valid pathogen test can be created. Missing sample and environmentSample links.");
```

Comment:
"Pathogen test can be created. Missing sample or environmentSample links"

Reply:
wrote "cannot" instead of "can". Updated the message.